tools/code_format/glint.sh

```diff
@@ -49,7 +49,7 @@ check_trailing_whitespace () {
 # TODO(phlax): add hash/diff only filter for faster change linting
 #      this would also make it feasible to add as a commit/push hook
 find_text_files () {
-    git grep --cached -Il '' | grep -vE "$NOLINT_RE"
+    git diff --cached --name-only --diff-filter=ACRMTU main | grep -vE "$NOLINT_RE"
```

Comment:
i like this idea, but a few points:

- in ci it should do a full scan
- it should work against changes in any branch not necessarily main
- the current way doesnt search in binary files (not that this matters a huge amount but would be good to keep that)

Reply:
For point one and two, we can pass some value to achieve it.
For point three, `git diff` does not search in binary files I think.

Reply:
Hi, is there a value about branch and CI env or not, if not, I will add them.

Reply:
ping @phlax